sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CreateMultipleVirtualMachinesAndBatchQueryStatus.java

```diff
@@ -0,0 +1,159 @@
+                    .withAddressSpace("10.0.0.0/16")
+                    .withSubnet(subNetName, "10.0.1.0/24")
+                    .create();
+
+            System.out.println("Batch creating Virtual Machines");
+            for (long i = 0; i < desiredVMCount; i++) {
+                String vmName = Utils.randomResourceName(azureResourceManager, "javascvm", 15);
+                vmNameList.add("'".concat(vmName).concat("'"));
+                azureResourceManager.virtualMachines()
+                        .define(vmName)
+                        .withRegion(region)
+                        .withExistingResourceGroup(rgName)
+                        .withExistingPrimaryNetwork(primaryNetwork)
+                        .withSubnet(subNetName)
+                        .withPrimaryPrivateIPAddressDynamic()
+                        .withoutPrimaryPublicIPAddress()
+                        .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_18_04_LTS)
+                        .withRootUsername(Utils.randomResourceName(azureResourceManager, "tirekicker", 15))
+                        .withSsh(Utils.sshPublicKey())
+                        .withNewDataDisk(50, 1, CachingTypes.READ_WRITE)
+                        .withSize(VirtualMachineSizeTypes.fromString("Standard_D2a_v4"))
+                        .beginCreate();
+            }
+
+            System.out.println("Use Azure Resource Graph to batch query the status(the number of provisioningState=Succeeded and Failed).");
+            StringBuilder succeededQuery = new StringBuilder();
+            succeededQuery.append("resources ")
+                    .append("| where (resourceGroup =~ ('").append(rgName).append("')) ")
+                    .append("| where (type =~ ('microsoft.compute/virtualmachines')) ")
+                    .append("| where (name in~ (").append(String.join(",", vmNameList)).append(")) ")
+                    .append("| where (properties['provisioningState'] =~ ('Succeeded')) ")
+                    .append("| project name,type,properties['provisioningState']");
+            StringBuilder failedQuery = new StringBuilder();
+
+            failedQuery.append("resources ")
+                    .append("| where (resourceGroup =~ ('").append(rgName).append("')) ")
+                    .append("| where (type =~ ('microsoft.compute/virtualmachines')) ")
+                    .append("| where (name in~ (").append(String.join(",", vmNameList)).append(")) ")
+                    .append("| where (properties['provisioningState'] =~ ('Failed')) ")
+                    .append("| project name,type,properties['provisioningState']");
+
+            while (total < desiredVMCount
+                    && resourceGraphManager.resourceProviders().resources(new QueryRequest()
+                    .withSubscriptions(Collections.singletonList(azureResourceManager.subscriptionId()))
+                    .withQuery(failedQuery.toString())).totalRecords() == 0) {
```

Comment:
I am not very happy with 2 queries. Basically you are wasting half of the quota.

Reply:
Fixed in the new version.